tests/test_fitparams.py

```diff
@@ -0,0 +1,40 @@
+import logging
+import os
+import unittest
+
+import astropy.units as u
+
+import pint.models as models
+from pint import fitter, toa
+from pinttestdata import datadir
+
+
+class TestFitparams(unittest.TestCase):
+    """Test whether unfreezing parameters that are not in the model will cause fitting to fail"""
+
+    @classmethod
+    def setUpClass(cls):
+        # J0613 is in equatorial
+        cls.parfile = os.path.join(datadir, "NGC6440E.par")
+        cls.model = models.get_model(cls.parfile)
+
+        cls.timfile = os.path.join(datadir, "NGC6440E.tim")
+        cls.toas = toa.get_TOAs(cls.timfile, usepickle=False)
+        cls.fitter = fitter.WLSFitter(cls.toas, cls.model)
```

Comment:
This test case demonstrates the problem with the API - one should be able to freeze or thaw parameters without needing to load TOAs and construct a Fitter.

Reply:
I'm confused..  You *can* freeze and thaw without a fitter instance.  .frozen is a model parameter attribute that you can just set:
```
n [1]: m.F0.frozen                                                             
Out[1]: False

In [2]: m.F0.frozen = True                                                      

In [3]: m.F0.frozen                                                             
Out[3]: True
```
Is there something wrong with that?

Reply:
But going back to my last comment, the *value* of .frozen really only matters in the context of a Fitter.

Reply:
Indeed, Patrick's notebooks have a lot of for loops full of getattr to freeze everything and then manually unfreeze a few things one by one. It would be useful to be able to just set free_params to the list of free parameters - but you can't do that without constructing a Fitter.

Reply:
Are you just saying that you would like a free_params() and/or freeze_params() method in the timing model instances?  I think that is completely fine.

But you can do it with a one liner now without a Fitter (m=model):
`for p in m.params:  getattr(m, p).frozen = False if p in ['F0', 'F1'] else True`

Reply:
Or if you want `free_params` to be a list of the free params, a list comp does it:
`free_params = [p for p in mm.params if getattr(mm, p).frozen==False]`

So I'm just confused why you say you need a fitter for these things?

If it is just that you want the syntactic sugar of set_free_params() or somesuch in the model, as I said, I'm totally fine with that!

Reply:
Since the `.frozen` feature of parameters lives in the model object, I don't want any manipulation of it, individually or in bulk, to have to go through a Fitter object - and `set_fitparams` does, it lives in a Fitter object. So I want to make that ability - set in bulk, which is featured in example notebooks because it's really convenient not to have to write a loop (let alone use `getattr`) - available from model objects directly. Which #822 does. I'd like to remove it from Fitter objects, which #822 also does, with a deprecation warning.

Reply:
I'd rather not have it removed from fitter objects.  I think it is useful there.  Fitter objects need a model, and they are the only places where frozen matters.

Once again, I'm completely happy to have the bulk setting and getting put into the timing model objects.  I just want the fitter to be able to access that directly.